src/osp/CoordinateSpaces/CartesianSimple.cpp

```diff
@@ -47,36 +47,30 @@ void CoordspaceCartesianSimple::update_exchange(
         rData.m_velocities.push_back(vel);
     }
 
-    rSpace.m_toAdd.clear();
-
     // TODO: handle removal of satellites
 }
 
 void CoordspaceCartesianSimple::update_views(
         CoordinateSpace& rSpace, CoordspaceCartesianSimple& rData)
 {
-
-    size_t const ccompCount = std::max({
-                ccomp_id<CCompSat>(),
-                ccomp_id<CCompX>(), ccomp_id<CCompY>(), ccomp_id<CCompZ>()});
-
     size_t const satCount = rData.m_satellites.size();
 
-    rSpace.m_components.resize(ccompCount + 1);
+    rSpace.m_components.resize(
+                ccomp_min_size<CCompSat, CCompX, CCompY, CCompZ>());
 
-    rSpace.m_components[ccomp_id<CCompSat>()] = {
+    rSpace.m_components[ccomp_id<CCompSat>] = {
                 rData.m_satellites, satCount,
                 sizeof(Satellite)};
 
-    rSpace.m_components[ccomp_id<CCompX>()] = {
+    rSpace.m_components[ccomp_id<CCompX>] = {
```

Comment:
I almost wonder if we should use the std:: namespace convention of using a _v suffix for template variables like this.